src/Symfony/Component/AssetMapper/Tests/Factory/MappedAssetFactoryTest.php

```diff
@@ -121,6 +122,32 @@ public function compile(string $content, MappedAsset $asset, AssetMapperInterfac
         $this->assertSame('7e4f24ebddd4ab2a3bcf0d89270b9f30', $asset->digest);
     }
 
+    /**
+     * @dataProvider provideHashAlgorithms
+     */
+    public function testCreateMappedAssetHashAlgorithms(?string $hashAlgorithm)
+    {
+        $factory = $this->createFactory(null, $hashAlgorithm);
+        $asset = $factory->createMappedAsset('subdir/file6.js', __DIR__.'/../Fixtures/dir2/subdir/file6.js');
+        $this->assertSame(hash_file($hashAlgorithm ?? 'xxh128', __DIR__.'/../Fixtures/dir2/subdir/file6.js'), $asset->digest);
```

Comment:
This test repeats the implementation under test. Can we test against actual hashes instead?

Reply:
👍  Updated!